Scheduler add getting started (#1484)
DevExpress/devextreme-documentation: concepts/05 Widgets/Scheduler/00 Getting Started with Scheduler/05 Bind the Scheduler to Data.md

The **Scheduler** can load and update data from these data source types:

> * Local array;

Reviewer: A list item should end with a full stop if it is a sentence. In all other cases, no punctuation marks are required.

* Read-Only Data in JSON Format;

* OData;

* Web API, PHP, MongoDB;

> * Custom data source.

Reviewer: The list items should lead to corresponding help topics.

> Use the [dataSource](/Documentation/ApiReference/UI_Widgets/dxScheduler/Configuration/#dataSource) option to specify a data source. In this tutorial, we will use a local array. 

Reviewer: Information about the ...Expr options is missing. It is better to use a data source with fields that differ from the default.

[note]The widget supports three types of appointments: one-time, all-day, and recurring. The used data source contains them all. 

---
##### jQuery

    <!-- tab: index.js --> 
    $(function() { 
        $("#scheduler").dx{WidgetName}({ 
            // ...
            dataSource: data
        });
    });

    <!-- tab: index.html -->
    <html>
        <head>
            <!-- ... -->
            <script src="data.js"></script>
        </head>
        <body class="dx-viewport">
            <div id="scheduler"></div>
        </body>
    </html>

    <!-- tab: data.js -->

>     const data = [

Reviewer: Use meaningful names for arrays, for example, `appointments`.

The **Scheduler** can load and update data from these data source types:

* [Local array](/Documentation/Guide/Data_Binding/Specify_a_Data_Source/Local_Array/)

> * [Read-Only Data in JSON Format](/Documentation/Guide/Data_Binding/Specify_a_Data_Source/Read-Only_Data_in_JSON_Format/)

Reviewer: ```suggestion
* [Read-only data in JSON format](/Documentation/Guide/Data_Binding/Specify_a_Data_Source/Read-Only_Data_in_JSON_Format/)
```

* [OData](/Documentation/Guide/Data_Binding/Specify_a_Data_Source/OData/)

* [Web API, PHP, MongoDB](/Documentation/Guide/Data_Binding/Specify_a_Data_Source/Web_API,_PHP,_MongoDB/)

> * [Custom data sources](/Documentation/Guide/Data_Binding/Specify_a_Data_Source/Custom_Data_Sources/).

Reviewer: ```suggestion
* [Custom data sources](/Documentation/Guide/Data_Binding/Specify_a_Data_Source/Custom_Data_Sources/)
```

> Use the [dataSource](/Documentation/ApiReference/UI_Widgets/dxScheduler/Configuration/#dataSource) property to specify a data source. In this tutorial, we will use a local array. Its objects have non-standard fields. To specify what standard field each custom one defines, we will use the following **Scheduler**'s properties: [textExpr](/Documentation/ApiReference/UI_Widgets/dxScheduler/Configuration/#textExpr), [startDateExpr](/Documentation/ApiReference/UI_Widgets/dxScheduler/Configuration/#startDateExpr), [endDateExpr](/Documentation/ApiReference/UI_Widgets/dxScheduler/Configuration/#endDateExpr), [allDayExpr](/Documentation/ApiReference/UI_Widgets/dxScheduler/Configuration/#allDayExpr), and [recurrenceRuleExpr](/Documentation/ApiReference/UI_Widgets/dxScheduler/Configuration/#recurrenceRuleExpr).

Reviewer: - First give information, then describe how it relates to the current case.
- Put more effort in your texts. "To specify what standard field each custom one defines" - this piece could use significant improvement.
- Let's show some fields with default names, some with custom.

```suggestion
Use the [dataSource](/Documentation/ApiReference/UI_Widgets/dxScheduler/Configuration/#dataSource) property to specify a data source. In this tutorial, we use a local array.

Your data source should contain data objects. If their fields match the [predefined structure](/Documentation/ApiReference/Common/Object_Structures/dxSchedulerAppointment/), the **Scheduler** recognizes them automatically and displays appointments without further configuration. Otherwise, you need to specify your field names in **...Expr** properties. In this tutorial, the `startDate` and `endDate` fields are recognized automatically, whereas other field names are specified in the [textExpr](/Documentation/ApiReference/UI_Widgets/dxScheduler/Configuration/#textExpr), [allDayExpr](/Documentation/ApiReference/UI_Widgets/dxScheduler/Configuration/#allDayExpr), and [recurrenceRuleExpr](/Documentation/ApiReference/UI_Widgets/dxScheduler/Configuration/#recurrenceRuleExpr) properties.
```

> [note]The UI component supports three types of appointments: one-time, all-day, and recurring. The used data source contains them all. 

Reviewer: Avoid specifying how many there is of something because this number could change.

```suggestion
The **Scheduler** supports one-time, all-day, and recurring [appointment types](/Documentation/Guide/Widgets/Scheduler/Appointments/Appointment_Types/). The data source below contains them all.
```

---
##### jQuery

    <!-- tab: index.js -->
    $(function() { 
        $("#scheduler").dxScheduler({

>             // ...

Reviewer: ```suggestion

```

            dataSource: appointments,

>             textExpr: "appointmentText",

Reviewer: Let's use `title` as the field name.

            startDateExpr: "start",
            endDateExpr: "end",

>             allDayExpr: "wholeDay",

Reviewer: Let's use `dayLong` as the field name.

            recurrenceRuleExpr: "recurrence",

>             currentView: "week"

Reviewer: Nothing about `currentView` yet.

            end: new Date("2021-05-23T10:00:00.000Z"),
            recurrence: "FREQ=WEEKLY;INTERVAL=2;COUNT=2"
        }, {
            appointmentText: "Meeting of Instructors",
            priorityId: 2,
            start: new Date("2021-05-24T10:00:00.000Z"),
            end: new Date("2021-05-24T11:15:00.000Z"),
            recurrence: "FREQ=DAILY;BYDAY=WE;UNTIL=20211001"
        }, {
            appointmentText: "Recruiting students",
            priorityId: 2,
            start: new Date("2021-05-25T08:00:00.000Z"),
            end: new Date("2021-05-25T09:00:00.000Z"),
            recurrence: "FREQ=YEARLY",
        }, {
            appointmentText: "Monthly Planning",
            priorityId: 2,
            start: new Date("2021-05-26T09:30:00.000Z"),
            end: new Date("2021-05-26T10:45:00.000Z"),
            recurrence: "FREQ=MONTHLY;BYMONTHDAY=28;COUNT=1"
        }, {
            appointmentText: "Open Day",
            priorityId: 2,
            start: new Date("2021-05-27T09:30:00.000Z"),
            end: new Date("2021-05-27T19:00:00.000Z"),
        }
    ];

##### Angular 

    <!-- tab: app.component.html -->

>     <dx-scheduler ...

Reviewer: ```suggestion
    <dx-scheduler
```

        [dataSource]="appointments"
        textExpr="appointmentText"
        startDateExpr="start"
        endDateExpr="end"
        allDayExpr="wholeDay"
        recurrenceRuleExpr="recurrence">
    </dx-scheduler> 

    <!-- tab: app.component.ts -->
    import { Component } from '@angular/core'; 
    import { Appointment, AppService } from './app.service';

    @Component({ 
        selector: 'app-root', 
        templateUrl: './app.component.html', 
        styleUrls: ['./app.component.css'],
        providers: [AppService]
    }) 
    export class AppComponent { 

>         // ...

Reviewer: ```suggestion
```

            appointmentText: "Meeting of Instructors",
            priorityId: 2,
            start: new Date("2021-05-24T10:00:00.000Z"),
            end: new Date("2021-05-24T11:15:00.000Z"),
            recurrence: "FREQ=DAILY;BYDAY=WE;UNTIL=20211001"
        }, {
            appointmentText: "Recruiting students",
            priorityId: 2,
            start: new Date("2021-05-25T08:00:00.000Z"),
            end: new Date("2021-05-25T09:00:00.000Z"),
            recurrence: "FREQ=YEARLY",
        }, {
            appointmentText: "Monthly Planning",
            priorityId: 2,
            start: new Date("2021-05-26T09:30:00.000Z"),
            end: new Date("2021-05-26T10:45:00.000Z"),
            recurrence: "FREQ=MONTHLY;BYMONTHDAY=28;COUNT=1"
        }, {
            appointmentText: "Open Day",
            priorityId: 2,
            start: new Date("2021-05-27T09:30:00.000Z"),
            end: new Date("2021-05-27T19:00:00.000Z"),
        }
    ];

    @Injectable({
        providedIn: 'root'
    })
    export class AppService {
        getAppointments(): Appointment[] {
            return appointments;
        }
    }

##### Vue 

    <!-- tab: App.vue -->
    <template> 

>         <DxScheduler ...

Reviewer: ```suggestion
        <DxScheduler
```

            :data-source="appointments"

>             textExpr="appointmentText"

Reviewer: ```suggestion
            text-expr="appointmentText"
```

            startDateExpr="start"
            endDateExpr="end"

>             allDayExpr="wholeDay"

Reviewer: ```suggestion
            all-day-expr="wholeDay"
```

>             recurrenceRuleExpr="recurrence">

Reviewer: ```suggestion
            recurrence-rule-expr="recurrence">
```

        </DxScheduler>
    </template> 

    <script>

> 

Reviewer: ```suggestion
```

    import { DxScheduler } from 'devextreme-vue/scheduler';
    import { appointments } from './data.js';

    export default {

>         name: 'Vue: Getting started with Scheduler',

Reviewer: ```suggestion

```

        components: {
            DxScheduler
        },
        data() {
            return {

>                 // ...

Reviewer: ```suggestion
```

            end: new Date("2021-05-24T11:15:00.000Z"),
            recurrence: "FREQ=DAILY;BYDAY=WE;UNTIL=20211001"
        }, {
            appointmentText: "Recruiting students",
            priorityId: 2,
            start: new Date("2021-05-25T08:00:00.000Z"),
            end: new Date("2021-05-25T09:00:00.000Z"),
            recurrence: "FREQ=YEARLY",
        }, {
            appointmentText: "Monthly Planning",
            priorityId: 2,
            start: new Date("2021-05-26T09:30:00.000Z"),
            end: new Date("2021-05-26T10:45:00.000Z"),
            recurrence: "FREQ=MONTHLY;BYMONTHDAY=28;COUNT=1"
        }, {
            appointmentText: "Open Day",
            priorityId: 2,
            start: new Date("2021-05-27T09:30:00.000Z"),
            end: new Date("2021-05-27T19:00:00.000Z"),
        }
    ];

##### React 

    <!-- tab: App.js -->
    import React from 'react';
    import 'devextreme/dist/css/dx.common.css';
    import 'devextreme/dist/css/dx.light.css';

    import { Scheduler } from 'devextreme-react/scheduler';
    import { appointments } from './data.js';

    function App() {
        return (
            <div className="App">

>                 <Scheduler ...

Reviewer: ```suggestion
                <Scheduler
```

            priorityId: 2,
            start: new Date("2021-05-23T09:00:00.000Z"),
            end: new Date("2021-05-23T10:00:00.000Z"),
            recurrence: "FREQ=WEEKLY;INTERVAL=2;COUNT=2"
        }, {
            appointmentText: "Meeting of Instructors",
            priorityId: 2,
            start: new Date("2021-05-24T10:00:00.000Z"),
            end: new Date("2021-05-24T11:15:00.000Z"),
            recurrence: "FREQ=DAILY;BYDAY=WE;UNTIL=20211001"
        }, {
            appointmentText: "Recruiting students",
            priorityId: 2,
            start: new Date("2021-05-25T08:00:00.000Z"),
            end: new Date("2021-05-25T09:00:00.000Z"),
            recurrence: "FREQ=YEARLY",
        }, {
            appointmentText: "Monthly Planning",
            priorityId: 2,
            start: new Date("2021-05-26T09:30:00.000Z"),
            end: new Date("2021-05-26T10:45:00.000Z"),
            recurrence: "FREQ=MONTHLY;BYMONTHDAY=28;COUNT=1"
        }, {
            appointmentText: "Open Day",
            priorityId: 2,
            start: new Date("2021-05-27T09:30:00.000Z"),
            end: new Date("2021-05-27T19:00:00.000Z"),
        }
    ];

---

> Run the code and ensure that the widget properly displays all appointments.

Reviewer: ```suggestion
Run the code and ensure that the **Scheduler** properly displays all appointments.
```
